twoliter/src/project/migrate/mod.rs

```diff
@@ -0,0 +1,586 @@
+//! Migration system for Twoliter project files.
+//!
+//! This module provides functionality to automatically migrate Twoliter.toml files between different schema versions,
+//! ensuring backward compatibility when new fields are added or the schema changes.
+
+use crate::{project::migrate::v1_to_v2::V1ToV2Migrator, schema_version::SupportedSchemaVersion};
+use anyhow::{anyhow, Context, Result};
+use std::any::Any;
+
+mod v1_to_v2;
+/// Trait for migrating project configurations between schema versions.
+pub trait ProjectMigrator: Send + Sync {
+    /// The source schema version this migrator handles.
+    fn current_version(&self) -> SupportedSchemaVersion;
+
+    /// The target schema version this migrator produces.
+    fn to_version(&self) -> SupportedSchemaVersion;
+
+    /// Perform the migration from source to target version.
+    fn migrate(&self, input: &dyn Any) -> Result<Box<dyn Any>>;
+
+    /// Parse the souce file to a target version
+    fn parse_source_version(&self, input: &str) -> Result<Box<dyn Any>>;
+}
+
+/// Registry of available project migrators.
+pub struct MigrationRegistry {
+    migrators: Vec<Box<dyn ProjectMigrator>>,
+}
+
+impl Default for MigrationRegistry {
+    fn default() -> Self {
+        MigrationRegistry::new(vec![Box::new(V1ToV2Migrator) as _])
+    }
+}
+
+impl MigrationRegistry {
+    /// Create a new migration registry with all available migrators.
+    pub fn new(migrators: Vec<Box<dyn ProjectMigrator>>) -> Self {
+        Self { migrators }
+    }
```

Comment:
We discussed in person: We probably want to detect duplicated migration intervals here. (e.g. don't have two migrations starting from `1`, don't have gaps of versions greater than 1.